### ISSUE TITLE
HeidiSQL does not currently work with Wine 7.4

### ISSUE THREAD

**Issue by kakenbok** [HeidiSQL/HeidiSQL#1551]
**Preconditions**

* HeidiSQL version: Version 11.3.0.6295 (64 Bit)
* Database type and version:
* OS: Linux, Ubuntu 21.10

**Describe the bug**

The error popup is raised upon startup. Many (almost all) features do not work any more.

**To Reproduce**

**Screenshots or Crash reports**

**Workaround**

Downgrade Wine to former version 7.3:

```
sudo apt-get install wine-devel=7.3~impish-1
sudo apt-get install wine-devel-amd64=7.3~impish-1
sudo apt-get install wine-devel-i386=7.3~impish-1
sudo apt-get install wine-devel=7.3~impish-1
sudo apt-get install winehq-devel=7.3~impish-1
```

(A problem is, that the system always wants to re-upgrade wine to 7.4 automatically.)

**Comment by direc85** [comment]
Same here with Manjaro Linux and Wine 7.4-1. Downgrading to 7.3-1 works.

**Comment by direc85** [comment]
Here is the HeidiSQL
[bugreport.txt](https://github.com/HeidiSQL/HeidiSQL/files/8253277/bugreport.txt) for one such crash: Clean install of Wine 7.4, portable installation of HeidiSQL 11.3.0.6295. I'll create a regression bug in WineHQ and cross-link them.

**Comment by ansgarbecker** [comment]
Could you please also update HeidiSQL to the latest build and try again. There are quite many changes after the last stable.
I will also release a new stable in the near future.

**Comment by direc85** [comment]
This Wine patch fixes this issue for me: https://bugs.winehq.org/show_bug.cgi?id=52678#c3
I applied on the master branch (currently 8a52d3e93f5690fd0833c228d54a996a784afad7)

**Comment by direc85** [comment]
Fixed in git: https://source.winehq.org/git/wine.git/commitdiff/d402df31966d73f7f120b4a4d90f22bcffd79b26

**Comment by ansgarbecker** [comment]
Thanks a lot for the deep look into this!
Does that mean the fix will be in the next Wine release?

**Comment by KOTRET** [comment]
The commit is part of the main branch, so I'd tend to say yes :) That was a quick fix

**Comment by kakenbok** [comment]
How can I get that patch until a new Wine version is released?

**Comment by direc85** [comment]
For example, Arch Linux has [wine-git](https://aur.archlinux.org/packages/wine-git) package which very conveniently fetches and compiles the latest sources. On other distribution there should be something similar. WineHQ also provides quite good instructions for building it yourself, if you have to go down that route.

**Comment by direc85** [comment]
The fix is now released in Wine 7.5.

**Comment by kakenbok** [comment]
can be closed
